### ISSUE TITLE
Any Plan or Suggestion for Wan2.1 LoRA Finetune?

### ISSUE THREAD

**Issue by BlossomsGarden** [mindspore-lab/mindone#1493]
Hi! Thanks for your valuable open-source works on NPU extention. 

I have noticed that you released LoRA-Finetune code for Wan-2.2-TI2V-5B. 

Do you have further plan on lora support on Wan-2.1-I2V-14B model?  

Or, can you offer some insights how  i can make it with a existing docker (mindspore=2.7.0rc1, cann=8.2rc1) on 910B? Like whether  it's feasible, which file i may edit, where i might create new files, etc..  

Thanks, have a good day :)

**Comment by vigo999** [comment]
we can support, may i ask what kind of experiment you are working on? the torch version lora is supported by DiffSynth-Studio 
https://github.com/modelscope/DiffSynth-Studio/tree/main/examples/wanvideo/model_training/lora
what kind of training code style do you want? based on vanilla scripts just for lora? or based on diffsynth lib?

**Comment by BlossomsGarden** [comment]
Thank you for your response! 

I aim to adapt the Wan2.1-I2V model for a video restoration/inpainting task. The goal is to take a distorted video + a mask video indicating the distorted regions + a clean initial frame as conditions, and have the model output the restored result.

<img width="2649" height="616" alt="Image" src="https://github.com/user-attachments/assets/6427552e-94de-4b8e-b8b6-f2433e6e416f" />

For me, vanilla scripts just for Wan2.1-I2V LoRA would be sufficient. My preliminary idea is to:

a) Prepare a dataset of (distorted video, mask video, initial frame, target restored video).

b) Train a new condition encoder to encode the distorted video and its mask into a conditioning vector

c) Fine-tune the Wan2.1 model with LoRA, using this new combined condition (from the new encoder + initial frame).

This is my initial thinking. I would greatly appreciate your professional advice on this plan:

- Does this approach seem feasible? 

- Are there any potential pitfalls or recommended best practices for my need? 

- Which key files in the current Wan codebase should I pay closest attention to when adapting the condition processing pipeline?

since I lack experience in finetuning large models, I'm actually not very certain if this is feasible 😂. i have also noticed comments about the difficulties of directly finetuning a 14B model. 

Thank you again for your time!

**Comment by wtomin** [comment]
@BlossomsGarden Hi, for docker image, I recommend you to look it up in https://quay.io/organization/ascend.

MindSpore docker image is a bit out of date. Please check in https://quay.io/repository/ascend/cann?tab=tags for your target CANN version. Afterwards, you can install mindspore=2.7.0rc1 by yourself.

**Comment by wtomin** [comment]
As for your proposed training method, I think it is feasible, but there are some risks of OOM and failure of convergence. Here are my suggestions:

- As it is a video inpainting task, you may also consider using the inverted noise of the distored video as the noisy input. In this case, you may not need to train a new condition encoder to encode the distorted video. The mask video is applied on the inverted noise. Any pixels that are masked out will be randomly initialized, while other pixels will be the inverted noise from the distorted video.
- If you want to train a new condition encoder, better to use a pretrained video encoder as initial weights. The video dataset size should be large enough for the new condition encoder to learn new visual features.
- Wan 2.1 (14B) maybe a little too large. How about finetuing your task with Wan-2.2-TI2V-5B first? If you can make it successfully converged on your data, then you can scale the model up with Wan-2.1-I2V-14B.

**Comment by BlossomsGarden** [comment]
Thank you very much for the detailed explanation and suggestions!

The idea of using the inverted noise from the distorted video is indeed clever and intuitive. Following your advice, I will first attempt to implement and experiment with this approach on the Wan-2.2-TI2V-5B model to verify the pipeline.

I appreciate your guidance and am excited to try this path.
